dotenv-linter/src/checks/value_without_quotes.rs

```diff
@@ -55,6 +61,8 @@ mod tests {
                 ("FOO=BAR BAZ", Some(WARNING)),
                 ("FOO=\"BAR BAZ\"", None),
                 ("FOO=\'BAR BAR\'", None),
+                ("FOO=BAR # Some Comment", None),
+                ("FOO=BAR BAZ # Some Comment", Some(WARNING)),
```

Comment:
Are your changes safe with something like this ?

```
FOO="Pas#word"
FOO="Pa #word"
```

Please add tests

Reply:
I have updated both the fix and the tests in order to cover these test cases with my recent commits